r/R/dplyr-select.R

```diff
@@ -32,6 +32,16 @@ rename.arrow_dplyr_query <- function(.data, ...) {
 }
 rename.Dataset <- rename.ArrowTabular <- rename.RecordBatchReader <- rename.arrow_dplyr_query
 
+rename_with.arrow_dplyr_query <- function(.data, .fn, .cols = everything(), ...) {
+  .fn <- rlang::as_function(.fn)
```

Comment:
In `arrow/r/R/arrow-package-R` we have a load of `rlang` function listed with `@importFrom` roxygen tags.  Could you add `as_function` there and then call it directly here?